patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -715,6 +724,15 @@
  			if (Main.npc[newNPC].type == 1 && Main.player[num7].RollLuck(180) == 0)
  				Main.npc[newNPC].SetDefaults(-4);
  
+@@ -62697,7 +_,7 @@
+ 		}
+ 
+ 		private static void GiveTownUniqueDataToNPCsThatNeedIt(int Type, int nextNPC) {
+-			if (TypeToDefaultHeadIndex(Type) != -1 || Type == 453) {
++			if (TypeToDefaultHeadIndex(Type) != -1 || Type == 453 || NPCLoader.ActsLikeATownNPC(Type)) {
+ 				Main.npc[nextNPC].GivenName = getNewNPCName(Type);
```

Comment:
Why does this add a check here? Terraria would use `isLikeATownNPC` instead of `Type == 453` otherwise

Reply:
I added the check here cause `isLikeATownNPC` is non-static and the method here is static, with the `Type` parameter coming from several other methods including `NewNPC`, thus requiring a `NPCLoader` call. `Type == 453 || NPCLoader.ActsLikeATownNPC(Type)` is equivalent to `isLikeATownNPC` as a result.

Reply:
`Main.npc[nextNPC].isLikeATownNPC` could work, but does it represent the same logic? Could require a different hook since vanilla would otherwise check that, but it didn't for some reason

Reply:
Apologies due to my lack of understanding, but is there something wrong with specifically the `NPCLoader` call being used here? I might be missing something obvious, and again sorry for my ignorance in this case.